internal/storage/payload_reader.go

```diff
@@ -413,6 +418,51 @@ func (r *PayloadReader) GetFloatVectorFromPayload() ([]float32, int, error) {
 	return ret, dim, nil
 }
 
+func (r *PayloadReader) GetSparseFloatVectorFromPayload() (*SparseFloatVectorFieldData, int, error) {
+	if !typeutil.IsSparseVectorType(r.colType) {
+		return nil, -1, fmt.Errorf("failed to get sparse float vector from datatype %v", r.colType.String())
+	}
+	values := make([]parquet.ByteArray, r.numRows)
+	valuesRead, err := ReadDataFromAllRowGroups[parquet.ByteArray, *file.ByteArrayColumnChunkReader](r.reader, values, 0, r.numRows)
+	if err != nil {
+		return nil, -1, err
+	}
+	if valuesRead != r.numRows {
+		return nil, -1, fmt.Errorf("expect %d binary, but got = %d", r.numRows, valuesRead)
+	}
+
+	fieldData := &SparseFloatVectorFieldData{}
+
+	for _, value := range values {
+		if len(value)%8 != 0 {
+			return nil, -1, fmt.Errorf("invalid bytesData length")
+		}
+		elementCount := len(value) / 8
+		indices := make([]int32, elementCount)
+		values := make([]float32, elementCount)
+		for i := 0; i < elementCount; i++ {
```

Comment:
is there a reason we don't use Proto.marshal(SparseFloatRow) but serialize by customized code?